src/app.controller.ts

```
@@ -9,4 +9,17 @@ export class AppController {
     return this.appService.getHello();
   }
+
+  @Get('weird')
+  getWeird(): string {
+    type WeirdType<T> = T extends string ? { value: T } : never;
+    const weirdValue: WeirdType<'weird'> = { value: 'weird' };
+
+    const castedValue = weirdValue as unknown as { value: string };
+
+    type UnionType = { value: string } | { error: string };
+    const unionValue: UnionType = castedValue as UnionType;
```

ESLint

[error] 'unionValue' is assigned a value but never used.

(@typescript-eslint/no-unused-vars)